smart-contracts-tutorial/first-contract.md

```diff
@@ -0,0 +1,153 @@
+# Tutorial on creating and running the simplest Smart Contract
+
+This first contract will be a one-player game where the player tries to guess a number. If they guess right, they win, otherwise, they lose. The game can be played any number of times and there is no actual stake. For the purpose of keeping things simple, the number is hardcoded to be `37`
+
+
+## Set up the environment for the contract
+
+Clone the [cosmic-swingset repository](https://github.com/Agoric/cosmic-swingset):
+```sh
+git clone git@github.com:Agoric/cosmic-swingset.git
+```
+
+Follow the instruction to [build and setup cosmic-swingset](https://github.com/Agoric/cosmic-swingset#build-from-source) (~20 minutes)
+
+
+## Write a contract
+
+```sh
+# Make sure you're in the cosmic-swingset root directory
+echo $PWD
+# => /maybe/home/you/code-projects/cosmic-swingset
+
+# (optional) create a git branch
+# git branch -c my-contract-tutorial
+
+# Create the contract file
+touch lib/ag-solo/vats/guess37-contract.js
+```
+
+**copy/paste the following code in the contract file:**
+```js
+function guess37Contract(terms, inviteMaker) {
+  const seat = harden({
+    guess(attempt) {
+      return attempt === 37 ? 'you win' : 'you lose';
+    },
+  });
+
+  return harden({
+    playerInvite: inviteMaker.make('player', seat),
+  });
+}
+
+export default guess37Contract.toString();
+```
+
+The contract is materialized by the source code of the `start` function. JavaScript makes it possible to retrieve the source code of a function by calling [toString](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/toString) on the function
+
+```md
+TODO a documentation of the contract format
+- Explain the contract host
+    - Explain `terms` and `inviteMaker` arguments
+- Explain the [provided top-level variables](https://github.com/Agoric/ERTP/blob/8f42af53581821bb16a9f4e3d62603833354ef8b/core/contractHost.js#L53-L63)
+- Explain Jessie
+```
+
+Replace the content of `lib/ag-solo/vats/vat-pixel.js` with the following:
```

Comment:
Add a note that we're doing this to save a bunch of configuration work that isn't relevant to this exercise. Promise to document the proper way to install a new contract later.

Reply:
https://github.com/Agoric/Documentation/issues/24